nb_sb_global.go

```diff
@@ -0,0 +1,186 @@
+	Name        string
+	Options     map[interface{}]interface{}
+	ExternalID  map[interface{}]interface{}
+	Connections []string
+	SSL         string
+	IPSec       bool
+}
+
+type SBGlobalTableRow struct {
+	UUID        string
+	Name        string
+	Options     map[interface{}]interface{}
+	ExternalID  map[interface{}]interface{}
+	Connections []string
+	SSL         string
+	IPSec       bool
+}
+
+func (odbi *ovndb) addGlobalTableRowImp(name string, table string) (*OvnCommand, error) {
+	namedUUID, err := newRowUUID()
+	if err != nil {
+		return nil, err
+	}
+	row := make(OVNRow)
+	row["name"] = name
+
+	if uuid := odbi.getRowUUID(table, row); len(uuid) > 0 {
+		return nil, ErrorExist
+	}
+
+	insertOp := libovsdb.Operation{
+		Op:       opInsert,
+		Table:    table,
+		Row:      row,
+		UUIDName: namedUUID,
+	}
+
+	operations := []libovsdb.Operation{insertOp}
+	return &OvnCommand{operations, odbi, make([][]map[string]interface{}, len(operations))}, nil
+}
+
+func (odbi *ovndb) nbGlobalAddImp(name string) (*OvnCommand, error) {
+	return odbi.addGlobalTableRowImp(name, tableNBGlobal)
+}
+
+func (odbi *ovndb) sbGlobalAddImp(name string) (*OvnCommand, error) {
+	return odbi.addGlobalTableRowImp(name, tableSBGlobal)
+}
+
+func (odbi *ovndb) delGlobalTableRowImp(name string, table string) (*OvnCommand, error) {
+	row := make(OVNRow)
+	row["name"] = name
+
+	rowUUID := odbi.getRowUUID(table, row)
+	if len(rowUUID) == 0 {
+		return nil, ErrorNotFound
+	}
+
+	condition := libovsdb.NewCondition("name", "==", name)
+	deleteOp := libovsdb.Operation{
+		Op:    opDelete,
+		Table: table,
+		Where: []interface{}{condition},
+	}
+	operations := []libovsdb.Operation{deleteOp}
+	return &OvnCommand{operations, odbi, make([][]map[string]interface{}, len(operations))}, nil
+}
+
+func (odbi *ovndb) nbGlobalDelImp(name string) (*OvnCommand, error) {
+	return odbi.delGlobalTableRowImp(name, tableNBGlobal)
+}
+
+func (odbi *ovndb) sbGlobalDelImp(name string) (*OvnCommand, error) {
+	return odbi.delGlobalTableRowImp(name, tableSBGlobal)
+}
+
+func (odbi *ovndb) nbGlobalSetOptionsImp(options map[string]string) (*OvnCommand, error) {
+	return odbi.globalSetOptionsImp(options, tableNBGlobal)
+}
+
+func (odbi *ovndb) nbGlobalGetOptionsImp() (map[string]string, error) {
+	return odbi.globalGetOptionsImp(tableNBGlobal)
+}
+
+func (odbi *ovndb) sbGlobalSetOptionsImp(options map[string]string) (*OvnCommand, error) {
+	return odbi.globalSetOptionsImp(options, tableSBGlobal)
+}
+
+func (odbi *ovndb) sbGlobalGetOptionsImp() (map[string]string, error) {
+	return odbi.globalGetOptionsImp(tableSBGlobal)
+}
+
+func (odbi *ovndb) globalSetOptionsImp(options map[string]string, table string) (*OvnCommand, error) {
+	if options == nil || table == "" {
+		return nil, fmt.Errorf("Invalid arguments passed to set options: table: %s, options:  %v", table, options)
+	}
+	mutatemap, err := libovsdb.NewOvsMap(options)
+	if err != nil {
+		return nil, err
+	}
+
+	uuid, err := func() (string, error) {
```

Comment:
Assumes that the NB_Global and SB_Global tables can have at most one row as per the spec.

Reply:
Can you add a comment about how nbctl/sbctl does as an example for ease to understand.